### DISCUSSION TITLE
Vue template "optional chaining" breaks syntax highlighting

### DISCUSSION THREAD

**Discussion by TheJaredWilcurt** [community/community#55674] [Pull Requests]
### Select Topic Area

Bug

### Body

Optional Chaining in JavaScript allows for shorter safety checks.

Instead of `a && a.b && a.b.c && a.b.c.d` you can just do `a?.b?.c?.d`.

In Vue 2, optional chaining did not work in the template section, but it *does* work in Vue 3:

```vue
<!-- Vue 2 -->
<template>
  <div v-if="user && user.id">{{ user.name }}</div>
</template>
<script>
import { getUser } from '@/networkCalls.js';
export default {
  name: 'GitHubExample',
  data: function () {
    return {
      user: null
    }
  },
  created: function () {
    this.user = await getUser();
  }
}
</script>
```

```vue
<!-- Vue 3 -->
<template>
  <div v-if="user?.id">{{ user.name }}</div>
</template>
<script>
import { getUser } from '@/networkCalls.js';
export default {
  name: 'GitHubExample',
  data: function () {
    return {
      user: null
    }
  },
  created: function () {
    this.user = await getUser();
  }
}
</script>
```

As you can see, the syntax highlighting breaks on the optional chaining in the template. This becomes a problem when doing reviews of PR's.

**Comment by emilburganov** [comment]
Optional chaining (?.) has recently appeared in js and is not supported by all browsers. Maybe it didn't exist in vue 2 days.
